src/validate.ts

```diff
@@ -26,18 +35,64 @@ function responseValidation(apiResponse: object, schema: object) {
 
 /**
  * Identifies the response type and validates the function based on the schemas.
- * @param apiResponse User defined api response
- * @param responseType User defined intent response
+ * @param req Request from one of the generate request functions.
+ * @param apiResponse User defined api response.
  * @return Errors from AJV validation, if any. Undefined otherwise.
  */
 export function validate(intentRequest: object, apiResponse: object) {
   const responseType = intentRequest['inputs'][0]['intent'];
   if (responseType === 'action.devices.SYNC') {
-    return responseValidation(apiResponse, SYNC_RESPONSE_SCHEMA);
+    // validate traits with attributes schema
+    const syncErrors : object[] = [];
+    const syncDevices = apiResponse['payload']['devices'];
+    const syncDevicesLength = syncDevices.length;
+    for (let i = 0; i < syncDevicesLength; i++) {
+      const traits = syncDevices[i]['traits'];
+      const traitsLength = syncDevices[i]['traits'].length;
+      const attributes = syncDevices[i]['attributes'];
+      if (attributes === undefined) {
```

Comment:
I believe the thing you would want to do in this case is validate `{}` against the trait attribute schema.